### PR TITLE
fix asus spidering

### PR DESCRIPTION
**Pull request by armijnhemel**
Apparently ASUS reworked their website. This PR should fix that.

### CODE REVIEW

**Comment by rhelmke** [comment]
Hey Armijn,

apologies. As we just recently migrated this repo into our organisation, we forgot to set up notifications. I'm going to check this PR in the near future. :-)

File: firmware/spiders/asus.py [hunk 1]
```diff
@@ -1,3 +1,4 @@
+import re
```
Comment (rhelmke) [inline firmware/spiders/asus.py:1]:
```suggestion
from re import findall
```
You can not know that, but we prefer pulling in the methods we use into the file's root namespace. If that is okay, please consider accepting my suggestion here :-)

File: firmware/spiders/asus.py [hunk 2]
```diff
@@ -34,7 +35,13 @@ class AsusSpider(Spider):
     ]
 
     def parse(self, response):
-        for url_redirect in set(response.xpath('//div[contains(@class, "ProductCardNormal")]//a/@href').getall()):
+        url_redirects = set()
+        header_scripts = set(response.xpath('//head//script/text()').getall())
+        for header in header_scripts:
+           if not '"url"' in header:
+               continue
+           url_redirects.update(re.findall('"url": "(https://[\w\d\-\_\./]+)"', header))
```
Comment (rhelmke) [inline firmware/spiders/asus.py:43]:
```suggestion
           url_redirects.update(findall(r'"url": "(https://[\w\d\-\_\./]+)"', header))
```
Can you please use the `r`-prefix for regex patterns? Also, we would strike the `re`-namespace due to the change above.

Comment (rhelmke) [inline firmware/spiders/asus.py:43]:
```suggestion
            if not '"url"' in header:
                continue
            url_redirects.update(re.findall('"url": "(https://[\w\d\-\_\./]+)"', header))
```
There is mixed indentation here.

**Review by rhelmke — CHANGES_REQUESTED** [review]
Hey, thank you so much for your contribution!

I tested your fix, and it works. There are a few minor changes I'd like to see in your code. Then I'm happy to pull in the changes.

Unfortunately, this fix can only be temporary, as I've noticed that the scraper does not pull in all firmware images. For example, images for the `TUF-AX5400` are missing. I am not sure why this happens. Of course we'd be grateful if you can fix this. Otherwise, we'll create an internal issue and check it out some time early in the next year.

Cheers and happy holidays

**Comment by armijnhemel** [comment]
> Hey, thank you so much for your contribution!
> 
> I tested your fix, and it works. There are a few minor changes I'd like to see in your code. Then I'm happy to pull in the changes.
> 
> Unfortunately, this fix can only be temporary, as I've noticed that the scraper does not pull in all firmware images. For example, images for the `TUF-AX5400` are missing. I am not sure why this happens. Of course we'd be grateful if you can fix this. Otherwise, we'll create an internal issue and check it out some time early in the next year.

I think this is because ASUS probably changed the format of the website. I need to look into this.

Personally I am also interested in any of the source code archives that are published so I am thinking of downloading those as well. I am not sure if you want to crawl these as well.

File: firmware/spiders/asus.py (hunk 2 above)
Comment (armijnhemel) [reply firmware/spiders/asus.py]: Urgh. Fixed.

**Comment by rhelmke** [comment]
> Personally I am also interested in any of the source code archives that are published so I am thinking of downloading those as well. I am not sure if you want to crawl these as well.

Are you refering to the GPL sources? We'd be interested in them as well. Maybe it is possible to include the feature in a non-invasive way that allows easy enabling/disabling.

**Review by rhelmke — COMMENTED** [review]
Hey Armijn, happy new year!

We are planning to verify each spider internally. Thus, we will probably rework the ASUS one as well. Are there any updates from your side? If not, I'd suggest we merge this PR now.

**Comment by armijnhemel** [comment]
> Hey Armijn, happy new year!
> 
> We are planning to verify each spider internally. Thus, we will probably rework the ASUS one as well. Are there any updates from your side? If not, I'd suggest we merge this PR now.

No updates here!

**Comment by rhelmke** [comment]
Alrighty, thanks for the feedback!

**Pull request merged**